### PR TITLE
Use chart version for images

### PR DESCRIPTION
**Pull request by AlexGodbehere**
Use the helper function to use the chart version for the image tags if not specified in the values file.

- Replaced all remaining static calls to images with the helper functions.

### CODE REVIEW

File: deploy/templates/identity/kdc.yaml [hunk 1]
```diff
@@ -48,7 +48,7 @@ spec:
         # Do as little work as possible as root... We need to chown the storage
         # as the default permissions will not be correct.
         - name: chown-storage
-          image: "{{ .Values.identity.identity.image.registry }}/{{ .Values.identity.identity.image.repository }}:{{ .Values.identity.identity.image.tag }}"
+          image: "{{ include "amrc-connectivity-stack.image-name" .Values.identity.identity }}"
```
Comment (amrc-benmorrow) [inline deploy/templates/identity/kdc.yaml:51]: Is there a reason these aren't using `amrc-connectivity-stack.image`, which includes `imagePullPolicy` from the values file?

File: deploy/templates/manager/manager.yaml [hunk 2]
```diff
@@ -38,7 +38,7 @@ spec:
         # Do as little work as possible as root... We need to chown the storage
         # as the default permissions will not be correct.
         - name: chown-storage
-          image: "{{ .Values.manager.image.registry }}/{{ .Values.manager.image.repository }}:{{ .Values.manager.image.tag }}-backend"
+          image: "{{ include "amrc-connectivity-stack.image-name" .Values.configdb }}-backend"
```
Comment (amrc-benmorrow) [inline deploy/templates/manager/manager.yaml:41]: `configdb`...

File: deploy/templates/manager/schema-import-cron.yaml [hunk 3]
```diff
@@ -29,7 +29,7 @@ spec:
                 name: krb5-conf
           containers:
             - name: cron
-              image: "{{ .Values.manager.image.registry }}/{{ .Values.manager.image.repository }}:{{ .Values.manager.image.tag }}-backend"
+              image: "{{ include "amrc-connectivity-stack.image-name" .Values.configdb }}-backend"
```
Comment (amrc-benmorrow) [inline deploy/templates/manager/schema-import-cron.yaml:32]: ...`configdb` to here

File: deploy/templates/_helpers.tpl [hunk 4]
```diff
@@ -39,7 +39,8 @@ Create chart name and version as used by the chart label.
 Define the image for a container.
 */}}
 {{- define "amrc-connectivity-stack.image-name" -}}
-{{- .image.registry }}/{{ .image.repository }}:{{ .image.tag }}
+{{- $defaultTag := default .Release.Version .image.tag }}
```
Comment (amrc-benmorrow) [inline deploy/templates/_helpers.tpl:42]: I think this is still scoped within the `.Values.*` passed in as context, so `.Release.Version` won't work. We need `$.Release.Version`.

I think it's also worth making this `coalesce .image.tag $.acs.defaultTag $.Release.Version` or something to allow a global override of the image tag. In particular I think a Flux HelmRelease pulling directly from a GitRepository results in a weird `.Release.Version` which won't work correctly.

**Review by amrc-benmorrow — CHANGES_REQUESTED** [review]

File: deploy/templates/identity/kdc.yaml (hunk 1 above)
Comment (AlexGodbehere) [reply deploy/templates/identity/kdc.yaml:51]: No. I needed to use the `image-name` one to include the suffix for the manager and it slipped through with them. I'll update now.

**Review by amrc-benmorrow — APPROVED** [review]

**Review by amrc-benmorrow — APPROVED** [review]

**Pull request merged**